### PR TITLE
Refactoring the Security page and section

### PR DESCRIPTION
**Pull request by vkhoroz**
## Readiness

* [x] Merge (pending reviews)
* [ ] Merge after _date or event_
* [ ] Draft

## Overview

I was trying to simply find the best place were to paste a link to the TUF offline keys management in scope of https://github.com/foundriesio/docs/pull/495.

The initial idea was to just add it to the TOC under Security.
But that turned out to produce an ugly looking sidebar. Until I realized that it already looks as ugly as it could when browsing.

So, I ended up re-structuring this whole thing, by trying to logically group related items and adding more context.
I'm not sure I fully like what I've got, but at the least now I can easily find any information of interest.

## Checklist

_Optional. Add a 'x' to steps taken._
_You can fill this out after opening the PR. "Did I..."_

* [x] Run spelling and grammar check, preferably with linter.
* [x] Avoid changing any header associated with a link/reference.
* [x] Step through instructions (or ask someone to do so).
* [x] Review for [wordiness](https://languagetool.org/insights/post/wordiness/)
* [x] Match tone and style of page/section.
* [x] Run `make linkcheck`.
* [x] View HTML in a browser to check rendering.
* [x] Use [semantic newlines](https://bobheadxi.dev/semantic-line-breaks/).
* [x] follow best practices for commits.
  * [x] Descriptive title written in the imperative.
  * [x] Include brief overview of QA steps taken.
  * [x] Mention any related issues numbers.
  * [x] End message with sign off/DCO line (`-s, --signoff`).
  * [x] Sign commit with your gpg key (`-S, --gpg-sign`).
  * [x] Squash commits if needed.
* [x] Request PR review by a technical writer and at least one peer.

## Comments

_Any thing else that a maintainer/reviewer should know._
_This could include potential issues, rational for approach, etc._

I'm not sure I **really want** to merge this.
Maybe, we are better with what we have now.
So, first of all I want to get some feedback on this restructuring, and then decide how to proceed.

Signed-off-by: Volodymyr Khoroz <volodymyr.khoroz@foundries.io>

### CODE REVIEW

**Comment by doanac** [comment]
Docs for 08de42ac564c43454d746a6412e62dfd11444d67 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1960/docs/artifacts/html/index.html

**Comment by doanac** [comment]
Docs for d41171606c5bbdabf01190af18d5bced85032c6e are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1961/docs/artifacts/html/index.html

**Comment by kprosise** [comment]
link-check failure can be ignored

**Comment by kprosise** [comment]
With the deleted pages, we will likely want to use redirects, though I can do so or assist with doing so if we decide to go with the changes. Things look good at my first pass, but I will be giving a more complete review later.

**Review by vanmaegima — APPROVED** [review]
Minor comments but LGTM.
Thanks for this, great improvements to this page!

**Review by angolini — APPROVED** [review]
It's nice to see some reorg on this page. I like the suggestions from Vanessa.

**Comment by doanac** [comment]
I think this is a net improvement. I'll defer to our doc experts.

File: source/reference-manual/security/security.rst [hunk 1]
```diff
@@ -3,76 +3,127 @@
...
+
+Security has multiple layers and dimensions.
+It starts all the way from booting the device to running software on it and connecting to cloud services.
+The FoundriesFactory® provides a set of features to target every aspect of your Factory security.
+
+Below sections focus on the following aspects:
+- how to securely connect your devices to Foundries.io™ cloud services.
+- how to securely boot your devices;
+- how to securely update firmware and software on your devices;
+- how to securely store secrets on your devices;
```
[the remark this answers is not in the document]
Comment (vkhoroz) [reply source/reference-manual/security/security.rst]: It was a _usual_ problem with the blank line. Sorry for that.
https://github.com/foundriesio/docs/pull/497/commits/4452c8490e776e399e097dfb1204bb53125089b5

**Comment by doanac** [comment]
Docs for bc7ae89ed925c271ca7841f800c5e4ea1e0f5c82 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1967/docs/artifacts/html/index.html

File: source/reference-manual/security/security.rst [hunk 2]
```diff
@@ -3,76 +3,127 @@
...
+
+   authentication-xilinx
+   secure-boot-imx-habv4
+   secure-boot-imx-ahab
+   secure-boot-stm32mp1
+   secure-boot-uefi
+   secure-machines
+   revoke-imx-keys
+   tee-on-versal-acap
+   factory-keys
```
[the remark this answers is not in the document]
Comment (vkhoroz) [reply source/reference-manual/security/security.rst]: Thanks! I split it into two parts as you suggested: https://github.com/foundriesio/docs/pull/497/commits/7321e93781303f51b04b23d6ad096f07853d009e

As for a separate section for `factory-keys`, I don't feel enough expert on the topic to write about that.
I will leave it up to the platform team to extend this in a separate PR.

**Comment by kprosise** [comment]
@vkhoroz just address the conflicting file merge issue, squash the commits, and then I can go ahead and merge.

**Comment by vkhoroz** [comment]
\+ rebased

**Comment by doanac** [comment]
Docs for 60b4b6cff44cef832251739163504922415acc65 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1970/docs/artifacts/html/index.html

**Comment by vkhoroz** [comment]
\+ rebased

There was one new link added for `ref-cert-rotation` in the `ota-security.rst`.
I moved it under the `ref-secure-cloud-services` section in `security.rst`, where it fits organically.

@doanac FYI

**Pull request merged**